src/solver/ts-generator/thermal.cpp

```diff
@@ -608,24 +612,48 @@ std::vector<Data::ThermalCluster*> getAllClustersToGen(Data::AreaList& areas,
 
 bool GenerateThermalTimeSeries(Data::Study& study,
                                std::vector<Data::ThermalCluster*> clusters,
-                               uint year,
-                               Solver::IResultWriter& writer)
+                               Solver::IResultWriter& writer,
+                               const std::string& savePath)
 {
     logs.info();
     logs.info() << "Generating the thermal time-series";
-    Solver::Progression::Task progression(study, year, Solver::Progression::sectTSGThermal);
 
-    auto* generator = new GeneratorTempData(study, progression, writer);
+    bool archive = (0 != (study.parameters.timeSeriesToArchive & Data::timeSeriesThermal));
 
-    generator->currentYear = year;
+    auto generator = std::make_unique<GeneratorTempData>(study);
 
     // TODO VP: parallel
     for (auto* cluster : clusters)
-        (*generator)(*cluster->parentArea, *cluster);
+    {
+        ThermalInterface clusterInterface(cluster);
+        (*generator)(*cluster->parentArea, clusterInterface);
+
+        if (archive)
+            writeResultsToDisk(study, writer, *cluster->parentArea, *cluster, savePath);
 
-    delete generator;
+        cluster->calculationOfSpinning();
+    }
 
     return true;
 }
 
+bool generateLinkTimeSeries(Data::Study& study,
+                            std::vector<Data::AreaLink*> links,
+                            Solver::IResultWriter& writer,
+                            const std::string& savePath)
+{
+    logs.info();
+    logs.info() << "Generating the links time-series";
+
+    auto generator = std::make_unique<GeneratorTempData>(study);
+
+    for (auto* link : links)
+    {
+        ThermalInterface clusterInterface(
+          link->tsGeneration, link->directCapacities, link->with->name);
+        (*generator)(*link->from, clusterInterface);
```

Comment:
- We need to distinguish direct & indirect capacities
- `link->directCapacities` and `link->indirectCapacities` should not be provided, use a temporary instead to make it clear that `link->directCapacities` and `link->indirectCapacities` won't be used after generating TS.
- We need to write generated TS to disk